homeassistant/components/gotify/__init__.py

```diff
@@ -0,0 +1,23 @@
+"""The gotify integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up gotify_temp from a config entry."""
+    hass.data[DOMAIN] = entry.as_dict()
+
+    await hass.async_create_task(
+        discovery.async_load_platform(hass, "notify", DOMAIN, {}, entry.as_dict())
+    )
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return True
```

Comment:
The GotifyNotificationService and `hass.data[DOMAIN][entry.entry_id]` should be removed.
For the service use: `hass.services.async_remove(...)`.